src/transformers/modeling_tf_utils.py

```diff
@@ -2157,11 +2158,15 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                 if from_pt and os.path.isfile(os.path.join(pretrained_model_name_or_path, WEIGHTS_NAME)):
                     # Load from a PyTorch checkpoint in priority if from_pt
                     archive_file = os.path.join(pretrained_model_name_or_path, WEIGHTS_NAME)
+                elif from_pt and os.path.isfile(os.path.join(pretrained_model_name_or_path, WEIGHTS_INDEX_NAME)):
+                    # Load from a sharded PyTorch checkpoint
+                    archive_file = os.path.join(pretrained_model_name_or_path, WEIGHTS_INDEX_NAME)
+                    is_sharded = True
```

Comment:
Nice addition, maybe we should also support loading from a remote sharded checkpoint with `from_pt=True`? (It should be its own PR if we decide to support this.)